### PR TITLE
Add AudioListener/AudioPanner DOM interfaces

### PR DESCRIPTION
**Pull request by Manishearth**
Seems to work.

I'll need some changes to the servo-media side to support the panner
node getters as well as the older `setPosition()`/etc APIs. I'll get to
those later.

r? @ferjm

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21502)

### CODE REVIEW

**Comment by Manishearth** [comment]
Tested, the following example works:

```html
<script type="text/javascript">
let ctx = new AudioContext();
let osc = ctx.createOscillator();
let panner = new PannerNode(ctx, {"coneOuterAngle": 0, "positionX": 100, "positionY": 0, "positionZ": 100, "refDistance": 100, "rolloffFactor": 0.01});
osc.connect(panner);
panner.connect(ctx.destination);
osc.start();
panner.positionX.linearRampToValueAtTime(-100, 0.2);
panner.positionX.linearRampToValueAtTime(100, 0.2);
panner.positionX.linearRampToValueAtTime(-100, 0.4);
panner.positionX.linearRampToValueAtTime(-100, 0.4);
panner.positionX.linearRampToValueAtTime(100, 0.6);
panner.positionX.linearRampToValueAtTime(-100, 0.6);
panner.positionX.linearRampToValueAtTime(100, 0.8);
panner.positionX.linearRampToValueAtTime(100, 0.8);
panner.positionX.linearRampToValueAtTime(-100, 1.0);
panner.positionX.linearRampToValueAtTime(100, 1.0);
panner.positionX.linearRampToValueAtTime(-100, 1.2);
panner.positionX.linearRampToValueAtTime(-100, 1.2);
panner.positionX.linearRampToValueAtTime(100, 1.4);
panner.positionX.linearRampToValueAtTime(-100, 1.4);
panner.positionX.linearRampToValueAtTime(100, 1.6);
panner.positionX.linearRampToValueAtTime(100, 1.6);
</script>
```

(should I check this in to tests/html?)

File: components/script/dom/audiolistener.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            AutomationRate::A_rate,
+            0.,       // default value
+            f32::MIN, // min value
+            f32::MAX, // max value
+        );
+        let position_z = AudioParam::new(
+            window,
+            context,
+            node,
+            ParamType::Position(ParamDir::Y),
```
Comment (ferjm) [inline components/script/dom/audiolistener.rs]: This should be `ParamDir::Z`

File: components/script/dom/audiolistener.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            AutomationRate::A_rate,
+            0.,       // default value
+            f32::MIN, // min value
+            f32::MAX, // max value
+        );
+        let forward_z = AudioParam::new(
+            window,
+            context,
+            node,
+            ParamType::Forward(ParamDir::Y),
```
Comment (ferjm) [inline components/script/dom/audiolistener.rs]: `ParamDir::Z`

File: components/script/dom/audiolistener.rs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            AutomationRate::A_rate,
+            1.,       // default value
+            f32::MIN, // min value
+            f32::MAX, // max value
+        );
+        let up_z = AudioParam::new(
+            window,
+            context,
+            node,
+            ParamType::Up(ParamDir::Y),
```
Comment (ferjm) [inline components/script/dom/audiolistener.rs]: `ParamDir::Z`

File: components/script/dom/baseaudiocontext.rs [hunk 4]
```diff
@@ -297,6 +304,15 @@ impl BaseAudioContextMethods for BaseAudioContext {
         })
     }
 
+    /// https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-destination
```
Comment (ferjm) [inline components/script/dom/baseaudiocontext.rs]: Change link to `https://webaudio.github.io/web-audio-api/#dom-baseaudiocontext-listener`, please

**Review by ferjm — COMMENTED** [review]
Looks good!

We need to add the [channelCount](https://dxr.mozilla.org/servo/source/components/script/dom/audionode.rs#207) and [channelCountMode](https://dxr.mozilla.org/servo/source/components/script/dom/audionode.rs#240) [constraints](https://webaudio.github.io/web-audio-api/#audionode-channelcount-constraints) for `PannerNode`.

Likewise, even though we do not support HRTF yet, we should also add the `autiomationRate` [constraints](https://webaudio.github.io/web-audio-api/#audioparam-automation-rate-constraints) for `PannerNode` and `AudioListener` AudioParams.

Thanks!

**Comment by Manishearth** [comment]
We don't have autiomationrate constraints, the automationrate is simply ignored (this code already exists in the media crate)

**Comment by Manishearth** [comment]
This should cause a bunch of new WPT failures since previously some tests didn't fully run at all, but that's okay.

At some point we need to switch gears to WPT and whittle those down, most of them aren't much work.

**Comment by Manishearth** [comment]
ah, missed a git commit -- amend

@bors-servo try=wpt

**Comment by Manishearth** [comment]
Added the remaining accessors and methods

r? @ferjm

**Comment by Manishearth** [comment]
@bors-servo try=wpt

support for the attributes and methods should reduce the WPT failures (and make most online examples work)

**Review by ferjm — APPROVED** [review]
LGTM! Thank you!

**Comment by Manishearth** [comment]
@bors-servo r=ferjm

 - needed to run update-manifest because I changed interfaces.html

**Comment by Manishearth** [comment]
@bors-servo retry

```
{"status": "PASS", "group": "default", "message": null, "stack": null, "subtest": "document.open should throw an InvalidStateError with XML document even when there is an active parser executing script", "test": "/html/webappapis/dynamic-markup-insertion/opening-the-input-stream/bailout-exception-vs-return-xml.window.html", "line": 48140, "action": "test_result", "expected": "FAIL"}
```

seems intermittent, let's find out

**Pull request merged**